gammapy/maps/base.py

```diff
@@ -127,6 +127,11 @@ def create(cls, **kwargs):
         from .hpxmap import HpxMap
         from .wcsmap import WcsMap
 
+        if 'binsz' in kwargs:
+            kwargs['binsz'] = kwargs['binsz'].to('deg').value
+        if 'width' in kwargs:
```

Comment:
As Christoph wrote this is not backward compatible and the unit-less call are likely common. so indeed change to `Quantity(kwargs['binsz'],'deg')`